Moggles.Domain/Application.cs

```diff
@@ -35,25 +35,25 @@ private bool DeployEnvExists(string name)
             return DeploymentEnvironments.Exists(e => string.Compare(e.EnvName, name, StringComparison.OrdinalIgnoreCase) == 0);
         }
 
-        public static Application Create(string appName, string defaultEnvironmentName, bool defaultToggleValueForEnvironment)
+        public static Application Create(string appName, string defaultEnvironmentName, bool defaultToggleValueForEnvironment, string username)
         {
             var app = new Application
             {
                 Id = Guid.NewGuid(),
                 AppName = appName
             };
-            app.AddDeployEnvironment(defaultEnvironmentName, defaultToggleValueForEnvironment);
+            app.AddDeployEnvironment(defaultEnvironmentName, defaultToggleValueForEnvironment, username);
```

Comment:
lets limit the use of this parameter strictly to the flow where a user flips the toggle on or off. If required at some point at creation time, you could hard code as the user name "created by system"